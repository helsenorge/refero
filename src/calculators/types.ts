import { Questionnaire, QuestionnaireResponse } from 'fhir/r4';

import { ClearAction } from '@/actions/newValue';

export type RunEnableWhenInput = {
  questionnaire: Questionnaire | null | undefined;
  questionnaireResponse: QuestionnaireResponse | null | undefined;
};
export type RunEnableWhenResult = ClearAction[];
