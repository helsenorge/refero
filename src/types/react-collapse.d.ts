declare namespace ReactCollapse {
  interface CollapseProps {
    isOpened?: boolean;
    className?: string;
    springConfig?: Array<Object>;
    onRest?: () => void;
    hasNestedCollapse?: boolean;
  }

  export class Collapse extends React.Component<CollapseProps, {}> {}
}

declare module 'react-collapse' {
  const collapse: typeof ReactCollapse.Collapse;
  export = collapse;
}
