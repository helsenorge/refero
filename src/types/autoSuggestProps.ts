export interface AutoSuggestProps {
  minSearchCharacters: number;
  typingSearchDelay: number;
}
