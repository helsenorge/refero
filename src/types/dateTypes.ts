export enum DateTimeUnit {
  Date = 'date',
  Hours = 'hours',
  Minutes = 'minutes',
}

export enum DatePickerFormat {
  HH = 'HH',
  mm = 'mm',
}

export enum DateFormat {
  ddMMyyyy = 'dd.MM.yyyy',
  dMMyyyy = 'd. MMMM yyyy',
  yyyyMMdd = 'yyyy-MM-dd',
  yyyyMM = 'yyyy-MM',
  MMMMyyyy = 'MMMM yyyy',
}

export interface TimeValues {
  hours: string;
  minutes: string;
}
