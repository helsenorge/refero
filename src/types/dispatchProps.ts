import { ThunkDispatch } from "redux-thunk";
import { NewValueAction } from "../actions/newValue";
import { GlobalState } from "../reducers";
import { Path } from "../util/refero-core";
import { Questionnaire, QuestionnaireResponse } from "./fhir";

export interface DispatchProps {
  dispatch: ThunkDispatch<GlobalState, void, NewValueAction>;
  mount: () => void;
  updateSkjema: (
    questionnaire: Questionnaire,
    questionnaireResponse?: QuestionnaireResponse,
    language?: string,
    syncQuestionnaireResponse?: boolean
  ) => void;
  path: Array<Path>;
}