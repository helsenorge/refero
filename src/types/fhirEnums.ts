import { Coding } from 'fhir/r4';
const questionnaire_enable_operator_NotEquals: Coding = {
  code: '!=',
  display: 'Not Equals',
  system: 'http://hl7.org/fhir/questionnaire-enable-operator',
};
const questionnaire_enable_operator_LessThan: Coding = {
  code: '<',
  display: 'Less Than',
  system: 'http://hl7.org/fhir/questionnaire-enable-operator',
};
const questionnaire_enable_operator_LessOrEquals: Coding = {
  code: '<=',
  display: 'Less or Equals',
  system: 'http://hl7.org/fhir/questionnaire-enable-operator',
};
const questionnaire_enable_operator_Equals: Coding = {
  code: '=',
  display: 'Equals',
  system: 'http://hl7.org/fhir/questionnaire-enable-operator',
};
const questionnaire_enable_operator_GreaterThan: Coding = {
  code: '>',
  display: 'Greater Than',
  system: 'http://hl7.org/fhir/questionnaire-enable-operator',
};
const questionnaire_enable_operator_GreaterOrEquals: Coding = {
  code: '>=',
  display: 'Greater or Equals',
  system: 'http://hl7.org/fhir/questionnaire-enable-operator',
};
const questionnaire_enable_operator_Exists: Coding = {
  code: 'exists',
  display: 'Exists',
  system: 'http://hl7.org/fhir/questionnaire-enable-operator',
};

export const QuestionnaireEnableOperator = {
  /**
   * True if whether at least no answer has a value that is equal to the enableWhen answer.
   */
  NotEquals: questionnaire_enable_operator_NotEquals,
  /**
   * True if whether at least no answer has a value that is less than the enableWhen answer.
   */
  LessThan: questionnaire_enable_operator_LessThan,
  /**
   * True if whether at least no answer has a value that is less or equal to the enableWhen answer.
   */
  LessOrEquals: questionnaire_enable_operator_LessOrEquals,
  /**
   * True if whether at least one answer has a value that is equal to the enableWhen answer.
   */
  Equals: questionnaire_enable_operator_Equals,
  /**
   * True if whether at least no answer has a value that is greater than the enableWhen answer.
   */
  GreaterThan: questionnaire_enable_operator_GreaterThan,
  /**
   * True if whether at least no answer has a value that is greater or equal to the enableWhen answer.
   */
  GreaterOrEquals: questionnaire_enable_operator_GreaterOrEquals,
  /**
   * True if whether an answer exists is equal to the enableWhen answer (which must be a boolean).
   */
  Exists: questionnaire_enable_operator_Exists,
};
export enum QuestionnaireItemEnableBehaviorCodes {
  ALL = 'all',
  ANY = 'any',
}

export enum QuestionnaireItemTypeCodes {
  GROUP = 'group',
  DISPLAY = 'display',
  BOOLEAN = 'boolean',
  DECIMAL = 'decimal',
  INTEGER = 'integer',
  DATE = 'date',
  DATETIME = 'dateTime',
}
