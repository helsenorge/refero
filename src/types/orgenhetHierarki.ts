export interface OrgenhetHierarki {
  OrgenhetId: number;
  Navn: string;
  EnhetType: EnhetType;
  EndepunktId: string | null;
  UnderOrgenheter: Array<OrgenhetHierarki> | null;
}

export enum EnhetType {
  Region = 1,
  Foretak = 2,
  Sykehus = 3,
  Klinikk = 4,
  Avdeling = 5,
  Seksjon = 6,
  Sengepost = 7,
  Poliklinikk = 8,
  Tjeneste = 9,
}
