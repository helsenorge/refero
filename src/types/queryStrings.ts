export interface QueryStringsInterface {
  MessageId: string;
  ExternalId: string;
  Query?: string;
  ContentType?: string;
}