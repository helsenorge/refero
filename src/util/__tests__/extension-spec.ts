import extensionConstants from '../../constants/extensions';
import { QuestionnaireItem, Extension } from '../../types/fhir';
import {
  getValidationTextExtension,
  getExtension,
  getPlaceholder,
  getMaxValueExtensionValue,
  getMinValueExtensionValue,
  getMinOccursExtensionValue,
  getMaxOccursExtensionValue,
  getMinLengthExtensionValue,
  getRepeatsTextExtension,
  getRegexExtension,
  getCopyExtension,
} from '../extension';

describe('extensions', () => {
  describe('getValidationTextExtension', () => {
    it('should return validation text', () => {
      const extension: Extension = {
        url: extensionConstants.VALIDATIONTEXT_URL,
        valueString: 'Du må fylle ut feltet',
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getValidationTextExtension(item)).toMatchSnapshot();
    });
  });

  describe('getExtension', () => {
    it('should return correct extension', () => {
      const extension: Extension = {
        url: extensionConstants.VALIDATIONTEXT_URL,
        valueString: 'Du må fylle ut feltet',
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getExtension(extensionConstants.VALIDATIONTEXT_URL, item)).toEqual(extension);
    });
  });

  describe('getPlaceholder', () => {
    it('should return placeholder', () => {
      const extension: Extension = {
        url: extensionConstants.ENTRY_FORMAT_URL,
        valueString: 'placeholdervalue',
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getPlaceholder(item)).toMatchSnapshot();
    });
  });

  describe('getMaxValueExtensionValue', () => {
    it('should return max value', () => {
      const valueInteger = 2;
      const extension: Extension = {
        url: extensionConstants.MAX_VALUE_URL,
        valueInteger,
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getMaxValueExtensionValue(item)).toMatchSnapshot();
    });
  });

  describe('getMinValueExtensionValue', () => {
    it('should return min value', () => {
      const valueInteger = 2;
      const extension: Extension = {
        url: extensionConstants.MIN_VALUE_URL,
        valueInteger,
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getMinValueExtensionValue(item)).toMatchSnapshot();
    });
  });

  describe('getMinOccursExtensionValue', () => {
    it('should return min occurs value', () => {
      const valueInteger = 2;
      const extension: Extension = {
        url: extensionConstants.MIN_OCCURS_URL,
        valueInteger,
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getMinOccursExtensionValue(item)).toMatchSnapshot();
    });
  });

  describe('getMaxOccursExtensionValue', () => {
    it('should return max occurs value', () => {
      const valueInteger = 2;
      const extension: Extension = {
        url: extensionConstants.MAX_OCCURS_URL,
        valueInteger,
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getMaxOccursExtensionValue(item)).toMatchSnapshot();
    });
  });

  describe('getMinLengthExtensionValue', () => {
    it('should return min length value', () => {
      const valueInteger = 2;
      const extension: Extension = {
        url: extensionConstants.MIN_LENGTH_URL,
        valueInteger,
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getMinLengthExtensionValue(item)).toMatchSnapshot();
    });
  });

  describe('getRepeatsTextExtension', () => {
    it('should return repeats text', () => {
      const extension: Extension = {
        url: extensionConstants.REPEATSTEXT_URL,
        valueString: 'repeats text',
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getRepeatsTextExtension(item)).toMatchSnapshot();
    });
  });

  describe('getRegexExtension', () => {
    it('should return regex', () => {
      const extension: Extension = {
        url: extensionConstants.REGEX_URL,
        valueString: 'regex',
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getRegexExtension(item)).toMatchSnapshot();
    });
  });

  describe('getCopyExtension', () => {
    it('should return extension', () => {
      const extension: Extension = {
        url: extensionConstants.Copy_EXPRESSION,
        valueString: 'Du må fylle ut feltet',
      } as Extension;
      const item: QuestionnaireItem = {
        linkId: '2.1',
        type: 'group',
        extension: [extension],
      };
      expect(getCopyExtension(item)).toEqual(extension);
    });
  });
});
