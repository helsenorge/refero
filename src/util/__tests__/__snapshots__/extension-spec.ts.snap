// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extensions getMaxOccursExtensionValue should return max occurs value 1`] = `2`;

exports[`extensions getMaxValueExtensionValue should return max value 1`] = `2`;

exports[`extensions getMinLengthExtensionValue should return min length value 1`] = `2`;

exports[`extensions getMinOccursExtensionValue should return min occurs value 1`] = `2`;

exports[`extensions getMinValueExtensionValue should return min value 1`] = `2`;

exports[`extensions getPlaceholder should return placeholder 1`] = `"placeholdervalue"`;

exports[`extensions getRegexExtension should return regex 1`] = `"regex"`;

exports[`extensions getRepeatsTextExtension should return repeats text 1`] = `"repeats text"`;

exports[`extensions getValidationTextExtension should return validation text 1`] = `"Du m√• fylle ut feltet"`;
